Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2220,30 +2225,41 @@ void RenderFlexibleBox::alignChildren(const Vector<LineContext>& lineContexts)
                 continue;
             
             ItemPosition position = alignmentForChild(flexItem.box);
+            LayoutUnit maxAscent = position == ItemPosition::LastBaseline ? lineContext.lastBaselineMaxAscent : lineContext.maxAscent;
             if (position == ItemPosition::Stretch)
                 applyStretchAlignmentToChild(flexItem.box, lineCrossAxisExtent);
             LayoutUnit availableSpace = availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box);
-            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse);
+            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse, childIsParticipatingInBaselineAlignment(flexItem.box));
```

Comment:
You could store `childIsParticipatingInBaselineAlignment(flexItem.box)` to avoid computing it multiple times.

Reply:
Since we no longer treat orthogonal elements as a special case (we need to synthesize the baselines instead), this argument is actually no longer needed. I went ahead and removed it from the code here.